### PR TITLE
Move baseline from Java 11 to Java 17

### PR DESCRIPTION
**Pull request by gsmet**
Creating as draft as I want a full CI run in my fork and because things are still in discussion.

We will need to check that Ecosystem CI is ready for this too, and probably a few other things like Quickstarts CI.

### CODE REVIEW

File: independent-projects/extension-maven-plugin/pom.xml [hunk 1]
```diff
@@ -34,6 +34,7 @@
         <javax.annotation-api.version>1.3.2</javax.annotation-api.version>
         <javax.inject.version>1</javax.inject.version>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <!-- Not sure exactly why but the Maven plugins need to be compiled with Java 11 -->
         <maven.compiler.target>11</maven.compiler.target>
         <maven.compiler.source>11</maven.compiler.source>
         <maven.compiler.release>11</maven.compiler.release>
```
Comment (gsmet) [inline independent-projects/extension-maven-plugin/pom.xml:40]: @aloubyansky there's something odd here. If I remove these lines and default to the parent Java 17, I end up with the following error when running the tests (basically anything trying to use the plugin fails):

(I have the same problem with the `quarkus-maven-plugin`, will post the exact error message below)

```
[ERROR]   ExtensionDescriptorMojoTest.shouldExecuteSimplePomCleanly:73->makeMojo:250->AbstractMojoTestCase.lookupConfiguredMojo:461->PlexusTestCase.lookup:205 » ComponentLookup com.google.inject.ProvisionException: Unable to provision, see the following errors:

1) No implementation for BootstrapWorkspaceProvider was bound.
  while locating ExtensionDescriptorMojo
  at ClassRealm[plexus.core, parent: null]
      \_ installed by: WireModule -> PlexusBindingModule
  while locating Mojo annotated with @Named("io.quarkus:quarkus-extension-maven-plugin:999-SNAPSHOT:extension-descriptor")

1 error

======================
Full classname legend:
======================
BootstrapWorkspaceProvider: "io.quarkus.maven.BootstrapWorkspaceProvider"
ExtensionDescriptorMojo:    "io.quarkus.maven.ExtensionDescriptorMojo"
Mojo:                       "org.apache.maven.plugin.Mojo"
Named:                      "com.google.inject.name.Named"
PlexusBindingModule:        "org.eclipse.sisu.plexus.PlexusBindingModule"
WireModule:                 "org.eclipse.sisu.wire.WireModule"
========================
End of classname legend:
========================
```

File: devtools/maven/pom.xml [hunk 2]
```diff
@@ -15,6 +15,11 @@
     <name>Quarkus - Maven Plugin</name>
 
     <properties>
+        <!-- Not sure exactly why but the Maven plugins need to be compiled with Java 11 -->
+        <maven.compiler.target>11</maven.compiler.target>
+        <maven.compiler.source>11</maven.compiler.source>
+        <maven.compiler.release>11</maven.compiler.release>
```
Comment (gsmet) [inline devtools/maven/pom.xml:21]: @aloubyansky If I drop these lines and default to the parent Java 17, I end up with the following error when anything tries to use this plugin:

```
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:999-SNAPSHOT:generate-code (default) on project quarkus-integration-test-grpc-descriptor-sets-parent: Execution default of goal io.quarkus:quarkus-maven-plugin:999-SNAPSHOT:generate-code failed: Unable to load the mojo 'generate-code' (or one of its required components) from the plugin 'io.quarkus:quarkus-maven-plugin:999-SNAPSHOT': com.google.inject.ProvisionException: Unable to provision, see the following errors:
[ERROR]
[ERROR] 1) No implementation for QuarkusBootstrapProvider was bound.
[ERROR]   while locating GenerateCodeMojo
[ERROR]   at ClassRealm[plugin>io.quarkus:quarkus-maven-plugin:999-SNAPSHOT, parent: ClassLoaders$AppClassLoader@251a69d7]
[ERROR]       \_ installed by: WireModule -> PlexusBindingModule
[ERROR]   while locating Mojo annotated with @Named("io.quarkus:quarkus-maven-plugin:999-SNAPSHOT:generate-code")
[ERROR]
[ERROR] 1 error
[ERROR]
[ERROR] ======================
[ERROR] Full classname legend:
[ERROR] ======================
[ERROR] ClassLoaders$AppClassLoader: "jdk.internal.loader.ClassLoaders$AppClassLoader"
[ERROR] GenerateCodeMojo:            "io.quarkus.maven.GenerateCodeMojo"
[ERROR] Mojo:                        "org.apache.maven.plugin.Mojo"
[ERROR] Named:                       "com.google.inject.name.Named"
[ERROR] PlexusBindingModule:         "org.eclipse.sisu.plexus.PlexusBindingModule"
[ERROR] QuarkusBootstrapProvider:    "io.quarkus.maven.QuarkusBootstrapProvider"
[ERROR] WireModule:                  "org.eclipse.sisu.wire.WireModule"
[ERROR] ========================
[ERROR] End of classname legend:
[ERROR] ========================
[ERROR] 
```

**Review by gsmet — COMMENTED** [review]
@aloubyansky I added two comments to your attention. There's something odd with the Maven plugins.

**Comment by gsmet** [comment]
@geoand @gastaldi @aloubyansky this should be ready now if you want to have a look.

Granted it's not the most exciting PR to review :).

**Review by geoand — APPROVED** [review]
Pretty massive, but looks fine on a quick scan, thanks!

**Review by aloubyansky — APPROVED** [review]
Thanks @gsmet!

**Review by gastaldi — APPROVED** [review]
LGTM. Great job!

**Pull request merged**